docs/blog/2022-07-15-nanoframework-networking-azure.md

```diff
@@ -0,0 +1,1067 @@
+        foreach (var val in vals)
+        {
+            ret += $"{val} ";
+        }
+
+        ret += "\r\n";
+    }
+
+    if (e.Context.Request.ContentLength64 > 0)
+    {
+
+        ret += $"Size of content: {e.Context.Request.ContentLength64}\r\n";
+        byte[] buff = new byte[e.Context.Request.ContentLength64];
+        e.Context.Request.InputStream.Read(buff, 0, buff.Length);
+        ret += $"Hex string representation:\r\n";
+        for (int i = 0; i < buff.Length; i++)
+        {
+            ret += buff[i].ToString("X") + " ";
+        }
+
+    }
+
+    WebServer.OutPutStream(e.Context.Response, ret);
+}
+```
+
+This API example is basic but as you get the method, you can choose what to do.
+
+As you get the URL, you can check for a specific controller called. And you have the parameters and the content payload!
+
+Example of a result with call:
+
+![result](./doc/POSTcapture.jpg)
+
+And more! Check [the complete example](https://github.com/nanoframework/Samples/tree/main/samples/Webserver) for more about this WebServer!
+
+## Using HTTPS
+
+You will need to generate a certificate and keys:
+
+```csharp
+X509Certificate _myWebServerCertificate509 = new X509Certificate2(_myWebServerCrt, _myWebServerPrivateKey, "1234");
+
+// X509 RSA key PEM format 2048 bytes
+        // generate with openssl:
+        // > openssl req -newkey rsa:2048 -nodes -keyout selfcert.key -x509 -days 365 -out selfcert.crt
+        // and paste selfcert.crt content below:
+        private const string _myWebServerCrt =
+@"-----BEGIN CERTIFICATE-----
+MORETEXT
+-----END CERTIFICATE-----";
+
+        // this one is generated with the command below. We need a password.
+        // > openssl rsa -des3 -in selfcert.key -out selfcertenc.key
+        // the one below was encoded with '1234' as the password.
+        private const string _myWebServerPrivateKey =
+@"-----BEGIN RSA PRIVATE KEY-----
+MORETEXTANDENCRYPTED
+-----END RSA PRIVATE KEY-----";
+
+using (WebServer server = new WebServer(443, HttpProtocol.Https)
+{
+    // Add a handler for commands that are received by the server.
+    server.CommandReceived += ServerCommandReceived;
+    server.HttpsCert = _myWebServerCertificate509;
+
+    server.SslProtocols = System.Net.Security.SslProtocols.Tls | System.Net.Security.SslProtocols.Tls11 | System.Net.Security.SslProtocols.Tls12;
+    // Start the server.
+    server.Start();
+
+    Thread.Sleep(Timeout.Infinite);
+}
+```
+
+> IMPORTANT: because the certificate above is not issued by a Certificate Authority it won't be recognized as a valid certificate. If you want to access the nanoFramework device with your browser, for example, you'll have to add the (CRT file)[WebServer.Sample\webserver-cert.crt] as a trusted one. On Windows, you just have to double click on the CRT file and then click "Install Certificate...".
+
+You can of course use the routes as defined earlier. Both will work, event or route with the notion of controller.
```

Comment:
This needs some clarification

Reply:
I rewrote fully the sentence, you can check it, it's pushed on the branch